runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -47,8 +48,15 @@
     private LocalDateTime startTime;
     private long startAddress;
     private long endAddress;
+    @Getter
     private long numEntries = 0;
     private long numBytes = 0;
+    // SMR entries total size falling on EXACTLY on the write size limit set will create problems
+    // Making sure there is 5% room, a safety guard to account for checkpoint entry overhead
+    // (metadata)
```

Comment:
Consider rewarding this. Maybe "Creates a batch that is 95% of the the max write size to accommodate for metadata overhead"

Reply:
Done